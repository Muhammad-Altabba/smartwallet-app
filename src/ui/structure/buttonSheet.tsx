import React from 'react'
import { ActionSheet } from './actionSheet'
import I18n from 'src/locales/i18n'
import { View, StyleSheet } from 'react-native'
import { JolocomButton } from './'
import strings from 'src/locales/strings'
import { fontMedium } from 'src/styles/typography'
import { Colors } from 'src/styles'

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
  },
  confirm: {
    flex: 2,
  },
  cancel: {
    flex: 1,
    borderStyle: 'solid',
    borderWidth: 2,
    borderColor: Colors.iBorderGray,
    marginLeft: 12,
  },
  confirmText: {
    fontFamily: fontMedium,
  },
})

interface Props {
  onConfirm: () => void
  onCancel: () => void
  show?: boolean
  cancelText?: string
  confirmText?: string
  disabledConfirm?: boolean
}

export const ButtonSheet = ({
  show = true,
  onConfirm,
  onCancel,
  disabledConfirm,
  confirmText = strings.CONFIRM,
  cancelText = strings.CANCEL,
}: Props) => {
  return (
    <ActionSheet showSlide={show}>
      <View style={styles.wrapper}>
        <JolocomButton
          textStyle={styles.confirmText}
          containerStyle={styles.confirm}
          disabled={!!disabledConfirm}
          onPress={onConfirm}
          text={I18n.t(confirmText)}
        />
        <JolocomButton
          containerStyle={styles.cancel}
          onPress={onCancel}
          text={I18n.t(cancelText)}
          transparent
        />
      </View>
    </ActionSheet>
  )
}
