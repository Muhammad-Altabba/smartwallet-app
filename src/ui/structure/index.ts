export { Wrapper } from './wrapper'
export { CardWrapper } from './cardWrapper'
export { JolocomButton } from './jolocomButton'
