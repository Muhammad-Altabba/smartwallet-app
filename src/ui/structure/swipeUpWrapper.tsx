import React from 'react'
import {
  GestureResponderEvent,
  PanResponder,
  PanResponderGestureState,
  View,
} from 'react-native'

interface Props {
  onSwipeUp: (gestureState: PanResponderGestureState) => void
}

export const SwipeUpWrapper: React.FC<Props> = props => {
  const { onSwipeUp } = props

  const swipeConfig = {
    isClickThreshold: 5,
    directionOffset: 110,
  }

  const isGestureClick = (gestureState: PanResponderGestureState) =>
    Math.abs(gestureState.dx) < swipeConfig.isClickThreshold &&
    Math.abs(gestureState.dy) < swipeConfig.isClickThreshold

  const isSwipeUp = (gestureState: PanResponderGestureState) =>
    gestureState.dy < 0 &&
    Math.abs(gestureState.dx) < swipeConfig.directionOffset

  const shouldSetResponder = (
    event: GestureResponderEvent,
    gestureState: PanResponderGestureState,
  ) => event.nativeEvent.touches.length === 1 && !isGestureClick(gestureState)

  const responderEnd = (
    event: GestureResponderEvent,
    gestureState: PanResponderGestureState,
  ) => {
    if (isSwipeUp(gestureState)) {
      onSwipeUp(gestureState)
    }
  }

  const panResponder = PanResponder.create({
    onStartShouldSetPanResponder: shouldSetResponder,
    onMoveShouldSetPanResponder: shouldSetResponder,
    onPanResponderRelease: responderEnd,
    onPanResponderTerminate: responderEnd,
  })

  return (
    <View {...props} {...panResponder.panHandlers}>
      {props.children}
    </View>
  )
}
