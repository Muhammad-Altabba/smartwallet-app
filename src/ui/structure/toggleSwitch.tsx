import React, { useState } from 'react'
import { StyleSheet, TouchableOpacity, Animated } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import { darkGrey } from '../../styles/colors'

const styles = StyleSheet.create({
  track: {
    justifyContent: 'center',
    width: 45,
    height: 30,
    borderRadius: 45,
  },
  toggle: {
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute',
    width: 26,
    height: 26,
    borderRadius: 13,
  },
  gradientWrapper: {
    width: '100%',
    height: '100%',
    borderRadius: 13,
  },
})

interface Props {
  value: boolean
  onToggle: () => void
  onGradient?: string[]
  offGradient?: string[]
  trackColor?: string
}

export const ToggleSwitch = (props: Props) => {
  const { onToggle, value, onGradient, offGradient, trackColor } = props

  const onGradientColors = onGradient || [
    'rgb(145, 25, 66)',
    'rgb(210, 45, 105)',
  ]
  const offGradientColors = offGradient || [
    'rgb(12, 12, 12)',
    'rgb(12, 12, 12)',
  ]
  const trackBackgroundColor = trackColor || darkGrey

  const offPosition = 2
  const onPosition = 17
  const switchPosition = value ? onPosition : offPosition

  const [positionValue] = useState(new Animated.Value(switchPosition))

  const onPress = () => {
    Animated.sequence([
      Animated.timing(positionValue, {
        toValue: value ? offPosition : onPosition,
        duration: 300,
      }),
    ]).start()
    onToggle()
  }

  return (
    <TouchableOpacity
      onPress={onPress}
      activeOpacity={0.8}
      style={{ ...styles.track, backgroundColor: trackBackgroundColor }}
    >
      <Animated.View
        style={{
          ...styles.toggle,
          transform: [{ translateX: positionValue }],
        }}
      >
        <LinearGradient
          style={styles.gradientWrapper}
          //NOTE locations prop breaks the gradient on iOS
          //locations={[0, 6]}
          start={{ x: 0, y: 0 }}
          end={{ x: 1, y: 0 }}
          colors={value ? onGradientColors : offGradientColors}
        />
      </Animated.View>
    </TouchableOpacity>
  )
}
