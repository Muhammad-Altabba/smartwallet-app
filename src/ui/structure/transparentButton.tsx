import React from 'react'
import {
  StyleSheet,
  Text,
  TouchableHighlight,
  TouchableHighlightProps,
  Platform,
} from 'react-native'
import { sandLight } from '../../styles/colors'
import { textLG, baseFontStyles } from '../../styles/typography'

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    borderColor: sandLight,
    borderWidth: 1,
    borderRadius: 4,
    paddingHorizontal: 30,
    margin: 10,
  },
  text: {
    color: sandLight,
    ...baseFontStyles,
    fontSize: textLG,
    paddingTop: Platform.OS === 'ios' ? 5 : 0,
  },
})

interface Props extends TouchableHighlightProps {
  text: string
}
export const TransparentButton: React.FC<Props> = props => {
  const { style, text, ...otherProps } = props
  return (
    <TouchableHighlight style={[styles.container, style]} {...otherProps}>
      <Text style={styles.text}>{text}</Text>
    </TouchableHighlight>
  )
}
