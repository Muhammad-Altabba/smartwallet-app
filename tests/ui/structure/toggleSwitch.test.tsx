import React from 'react'
import { shallow } from 'enzyme'
import { ToggleSwitch } from '../../../src/ui/structure/toggleSwitch'
import { TouchableOpacity } from 'react-native'

describe('ToggleSwitch Component', () => {
  const defaultProps = {
    value: false,
    onToggle: jest.fn(),
  }

  it('matches the snapshot on initial render', () => {
    const component = shallow(<ToggleSwitch {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })

  it('renders the switched on toggle', () => {
    const props = { ...defaultProps, value: true }
    const component = shallow(<ToggleSwitch {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('renders the toggle with custom gradients', () => {
    jest.useFakeTimers()
    const props = {
      ...defaultProps,
      onGradient: ['rgb(13, 24, 156)', 'rgb(55, 34, 222)'],
      offGradient: ['rgb(53, 39, 126)', 'rgb(65, 24, 24)'],
    }
    const component = shallow(<ToggleSwitch {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('renders the toggle with custom track color', () => {
    jest.useFakeTimers()
    const props = { ...defaultProps, trackColor: 'rgb(0, 0, 0)' }
    const component = shallow(<ToggleSwitch {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('calls onToggle when pressed', () => {
    const component = shallow(<ToggleSwitch {...defaultProps} />)
    component.find(TouchableOpacity).simulate('press')
    expect(defaultProps.onToggle).toBeCalledTimes(1)
  })
})
