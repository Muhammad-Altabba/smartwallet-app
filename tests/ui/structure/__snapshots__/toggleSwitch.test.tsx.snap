// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch Component matches the snapshot on initial render 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(55, 53, 55)",
      "borderRadius": 45,
      "height": 30,
      "justifyContent": "center",
      "width": 45,
    }
  }
>
  <AnimatedComponent
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 13,
        "height": 26,
        "justifyContent": "center",
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 2,
          },
        ],
        "width": 26,
      }
    }
  >
    <LinearGradient
      colors={
        Array [
          "rgb(12, 12, 12)",
          "rgb(12, 12, 12)",
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 0,
        }
      }
      start={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      style={
        Object {
          "borderRadius": 13,
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </AnimatedComponent>
</TouchableOpacity>
`;

exports[`ToggleSwitch Component renders the switched on toggle 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(55, 53, 55)",
      "borderRadius": 45,
      "height": 30,
      "justifyContent": "center",
      "width": 45,
    }
  }
>
  <AnimatedComponent
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 13,
        "height": 26,
        "justifyContent": "center",
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 17,
          },
        ],
        "width": 26,
      }
    }
  >
    <LinearGradient
      colors={
        Array [
          "rgb(145, 25, 66)",
          "rgb(210, 45, 105)",
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 0,
        }
      }
      start={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      style={
        Object {
          "borderRadius": 13,
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </AnimatedComponent>
</TouchableOpacity>
`;

exports[`ToggleSwitch Component renders the toggle with custom gradients 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(55, 53, 55)",
      "borderRadius": 45,
      "height": 30,
      "justifyContent": "center",
      "width": 45,
    }
  }
>
  <AnimatedComponent
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 13,
        "height": 26,
        "justifyContent": "center",
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 2,
          },
        ],
        "width": 26,
      }
    }
  >
    <LinearGradient
      colors={
        Array [
          "rgb(53, 39, 126)",
          "rgb(65, 24, 24)",
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 0,
        }
      }
      start={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      style={
        Object {
          "borderRadius": 13,
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </AnimatedComponent>
</TouchableOpacity>
`;

exports[`ToggleSwitch Component renders the toggle with custom track color 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(0, 0, 0)",
      "borderRadius": 45,
      "height": 30,
      "justifyContent": "center",
      "width": 45,
    }
  }
>
  <AnimatedComponent
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 13,
        "height": 26,
        "justifyContent": "center",
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 2,
          },
        ],
        "width": 26,
      }
    }
  >
    <LinearGradient
      colors={
        Array [
          "rgb(12, 12, 12)",
          "rgb(12, 12, 12)",
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 0,
        }
      }
      start={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      style={
        Object {
          "borderRadius": 13,
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </AnimatedComponent>
</TouchableOpacity>
`;
